src/instructlab/cli/data/list.py

```diff
@@ -39,5 +67,34 @@ def list_datasets(dataset_dirs):
         click.secho(f"Failed to list datasets with exception: {exc}")
         raise click.exceptions.Exit(1)
 
+    grouped_data = defaultdict(list)
+    timestamp_to_model = {}
+    general_files = []
+
+    # Process model name files e.g. start with test_, train_, messages_
+    for item in data:
+        filename = item[0]
+        group_key, is_general, timestamp = extract_model_and_timestamp(filename)
+
+        if group_key:
+            if is_general:
+                general_files.append((item, timestamp))
+            else:
+                model_name = group_key.split(" ", 1)[0]
+                timestamp_to_model[timestamp] = f"{model_name} {timestamp}"
+                grouped_data[f"{model_name} {timestamp}"].append(item)
+
+    # Process General files e.g. start with skills_train_msgs_, node_datasets_
+    for item, timestamp in general_files:
+        model_group = timestamp_to_model.get(timestamp, f"General {timestamp}")
```

Comment:
When is the model_group not found in timestamp_to_model?  Comment might be helpful here.